### PR TITLE
Raytac mdbt50 q rx

### PR DESCRIPTION
**Pull request by pyro9**
Add support for Raytac MDBT50Q - RX USB dongle based on nrf52840.

### CODE REVIEW

File: variants/MDBT50Q_RX/variant.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+#include "variant.h"
+#include "wiring_constants.h"
+#include "wiring_digital.h"
+#include "nrf.h"
+
+const uint32_t g_ADigitalPinMap[] =
+{
+  // D0 .. D13
+  45,  // D3  is P1.13 (LED1)
```
Comment (hathach) [inline variants/MDBT50Q_RX/variant.cpp]: comment `D3 is P1.13 (LED1)` is out dated, should be D0 here, make sure you update its usage as well

**Comment by ladyada** [comment]
@hathach please order some dongles from raytac so we can test it out!

**Comment by hathach** [comment]
> @hathach please order some dongles from raytac so we can test it out!

OK, I will order a couple of them

**Comment by hathach** [comment]
@pyro9 I got the dongle, but don't know how to program or flash it. It comes with raytac firmware, but there is very little information. Would you mind giving me some help here. We will need to get our bootloader on it first, before able to use it with the arduino repo. 
https://www.raytac.com/product/ins.php?index_id=89

**Comment by pyro9** [comment]
> @pyro9 I got the dongle, but don't know how to program or flash it. It comes with raytac firmware, but there is very little information. Would you mind giving me some help here. We will need to get our bootloader on it first, before able to use it with the arduino repo.
> https://www.raytac.com/product/ins.php?index_id=89

Sorry about the delay getting back to you.
Refer to the datasheet: https://www.raytac.com/download/index.php?index_id=40
Chapter 9, page 21 has drawings showing the layout of the programming pads on the board. I made a programming cradle from a 3D printed part, pogo pins, and the Adafruit programming breakout board and plugged in the J-Link debugger.

I used the bootloader and soft device image for the Feather Express as-is (noting that the DFU button won't work) to get the board up and running, then loaded my software using the Arduino IDE as usual.

I found that Gorilla Glue would not properly adhere the pogo pins to the PLA I printed the cradle from, so I needed to use hot-melt after I soldered leads to the pins.

If you want FreeCAD, step, or stl for the cradle, let me know.

**Comment by hathach** [comment]
thanks, I guess I have to do a bit of soldering with those pad first, then I will the port for bootloader and check back this PR afterwards.

**Comment by hathach** [comment]
update: sorry for being late, but I will definitely close this PR this week :)

**Pull request merged**